File: preassignment/pre.py
import numpy as np
import matplotlib.pyplot as plt
import time
from tqdm import trange

N =200000 # Number of random walks
n = 200 # Max number of steps / shape of t-array
t = np.arange(n)

rand = np.random.uniform(-1,1,(N,n))
random_walks = np.cumsum(rand,axis = 1) - rand

p = np.zeros(n)

for i in trange(N,desc="1"):
    for j in range(n):
        if random_walks[i,1] > 0:
            if random_walks[i,j] <0:
                p[j] += 1
                break
        if random_walks[i,1] < 0:
            if random_walks[i,j] >0:
                p[j] += 1
                break

logt = np.log(t)[2:]
alpha = np.log(p[2:])
print(logt[:2])

coeffs = np.polyfit(logt, alpha,1)
print(logt*coeffs[0]  + coeffs[1])
print(alpha)
print(coeffs)

plt.figure()
plt.subplot(121)
plt.plot(p[2:], label = "Probability")

plt.subplot(122)
plt.plot(alpha, label = "Alpha")
plt.plot(logt*coeffs[0] + coeffs[1], label = "Regression")
#plt.xscale("log")

plt.legend()
plt.show()

# for i in tqdm(range(N)):

File: preassignment/pre.jl
using Random
using Plots;pyplot()

N=1000000
n= 100
t = [1:n]

r_dist =  randn!(MersenneTwister(), zeros(N,n))
println(typeof(r_dist)) # Array{Float64,2}
println(size(r_dist))
@time random_walks = cumsum(r_dist,dims = 2)
println(size(random_walks))

println(typeof(random_walks))
@time function checkstuff(index::Int,prob,walk::Array{Float64,2})
    for j=1:n
        if walk[index,1] >0
            if random_walks[index,j] <0
                prob[j] += 1
                return
            end
        elseif walk[index,1] <0
            if random_walks[index,j] >0
                prob[j] += 1
                return
            end
        end
    end
    return
end

p = zeros(n)

@time for i=1:N
    checkstuff(i,p,random_walks)
end

a = plot(p)
display(a)
readline()
